packages/eventindexer/indexer/save_block_proven_event.go

```diff
@@ -75,30 +74,35 @@ func (svc *Service) saveBlockProvenEvent(
 	eventindexer.BlockProvenEventsProcessed.Inc()
 
 	if event.Prover.Hex() != systemProver.Hex() && event.Prover.Hex() != oracleProver.Hex() {
-		if err := svc.updateAverageBlockTime(ctx, event); err != nil {
+		if err := svc.updateAverageProofTime(ctx, event); err != nil {
 			return errors.Wrap(err, "svc.updateAverageBlockTime")
 		}
 	}
 
 	return nil
 }
 
-func (svc *Service) updateAverageBlockTime(ctx context.Context, event *taikol1.TaikoL1BlockProven) error {
+func (svc *Service) updateAverageProofTime(ctx context.Context, event *taikol1.TaikoL1BlockProven) error {
 	block, err := svc.taikol1.GetBlock(nil, event.Id)
 	if err != nil {
 		return errors.Wrap(err, "svc.taikoL1.GetBlock")
 	}
 
+	eventBlock, err := svc.ethClient.BlockByHash(ctx, event.Raw.BlockHash)
+	if err != nil {
+		return errors.Wrap(err, "svc.ethClient.BlockByHash")
+	}
+
 	stat, err := svc.statRepo.Find(ctx)
 	if err != nil {
 		return errors.Wrap(err, "svc.statRepo.Find")
 	}
 
 	proposedAt := block.ProposedAt
 
-	provenAt := time.Now().Unix()
+	provenAt := eventBlock.Time()
```

Comment:
hmmm seems `eventBlock.Time()` should equal to `block.ProposedAt`

Reply:
i dont think so, `eventBlock.TIme()` is the time at which the L1 block was made with the `BlockProven` event with this Id inside of it.

`block.ProposedAt` is when the original block was proposed. 

eventBlock.Time() is greater than block.proposedAt.

Reply:
i dont think so, `eventBlock.TIme()` is the time at which the L1 block was made with the `BlockProven` event with this Id inside of it.

`block.ProposedAt` is when the original block was proposed. 

eventBlock.Time() is greater than block.proposedAt.